lib/recurly/pricing/subscription/tiered-pricing-calculator.js

```diff
@@ -0,0 +1,141 @@
+/* eslint-disable no-param-reassign */
+
+/**
+ * 
+ * Takes in an addon and returns boolean to decide if addOn should used tiered math
+ * 
+ * @param {object} addOn 
+ * @returns {boolean}
+ */
+export function isTieredAddOn (addOn) {
+  return addOn.add_on_type === 'fixed' && addOn.tier_type !== 'flat';
+}
+
+/**
+ *
+ * Calculates amount on each tier row depending on isPercentage param and currency precision.
+ *
+ * When isPercentage = true (tiered_usage_percentage_add_ons)
+ *
+ * @param {number} unitsPerTier
+ * @param {number} tierValue
+ * @returns {number}
+ *
+ */
+function calculateTotal (unitsPerTier, tierValue) {
+  return unitsPerTier * tierValue;
```

Comment:
Thinking we can collapse this operation into the caller since it's been simplified to the one mult op now.